private/buf/buflsp/report.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file provides helpers for bridging protocompile and LSP diagnostics.
+
+package buflsp
+
+import (
+	"fmt"
+
+	"github.com/bufbuild/protocompile/linker"
+	"github.com/bufbuild/protocompile/parser"
+	"github.com/bufbuild/protocompile/reporter"
+	"go.lsp.dev/protocol"
+)
+
+// report is a reporter.Reporter that captures diagnostic events as
+// protocol.Diagnostic values.
+type report struct {
```

Comment:
Since this implements `reporter.Reporter`, it should probably be named `reporter`.

Reply:
Instead of alias-ing the package, let's leave this as `report` -- it is still a clear name for the implementation, since it is a "report" of the diagnostics, missing syntax, and unused imports.

We should probably rename `syntaxMissing` and `unusedImports` with the same map naming conventions, so in this case `fileNameToSyntaxMissing` and `fileNameToUnusedImports`.

Reply:
Chatted in person, `syntaxMissing` is a good name for the set, and we'll rename `unusedImports`.